File: Leetcode (Python, JavaScript)/234. Palindrome Linked List/234. Palindrome Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        li = []
        curr = head
        while curr:
            li.append(curr.val)
            curr = curr.next
        if li==li[::-1]:
            return True
        else:
            return False

File: Leetcode (Python, JavaScript)/234. Palindrome Linked List/234. Palindrome Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function(head) {
    let li = []
    let curr = head;
    while (curr!==null){
        li.push(curr.val);
        curr = curr.next;
    }
    let reversedli = [...li].reverse();
    return li.join(',') === reversedli.join(',');
};